dev_container/compile_docker.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+
+BASE_IMAGE="ubuntu:jammy-20220421"
+
+IMAGE_NAME="scons_image"
+CONTAINER_NAME="scons_container"
+
+# local directories 
+C_CODE=$(cd ../ && pwd)
+WORKDIR=/usr/project/
+
+# wrapper variables 
+COMPONENT=""
+
+# TODO: find dockerfile --> will fail if Dockerfile not in directory script is executed from 
+
+# check for existence of image 
+
+docker images | grep ${IMAGE_NAME}
+if [[ $? -eq 0 ]]; then
+	echo "Image already exists. Skipping build."
+else 
+	echo "No image found. Pulling and building"
+	 
+	# pull base image 
+	docker pull ${BASE_IMAGE}
+	
+	# build base ubuntu image  
+	docker build -t ${IMAGE_NAME} .
+fi
+
+# start and mount program files to container workdir 
+docker run -t -d -v ${C_CODE}:${WORKDIR} --name ${CONTAINER_NAME} ${IMAGE_NAME}
+
+# install requirements 
+docker exec ${CONTAINER_NAME} pip3 install -r ${WORKDIR}/requirements.txt
+
+# rename and move compiler files to appropriate dir 
+docker exec ${CONTAINER_NAME} mv /gcc-arm-11.2-2022.02-x86_64-arm-none-eabi/ /usr/project/embedded/platforms/toolchain-gccarmnoneeabi/
```

Comment:
download the compiler in this script instead of in the Dockerfile, then you can extract here

also use `tar -xf <archive> -C embedded/toolchains/gcc-arm-none-eabi`